src/hearings/containers/request-hearing/hearing-change-reason/hearing-change-reason.component.ts

```diff
@@ -1,23 +1,116 @@
+      super(hearingStore, hearingsService);
+    }
+
+  public ngOnInit(): void {
+    this.hearingStore.pipe(select(fromHearingStore.getHearingList)).subscribe(
+      hearingList => {
+        this.caseId = hearingList.hearingListMainModel ? hearingList.hearingListMainModel.caseRef : '';
+        if (hearingList.hearingListMainModel) {
+          const caseHearings = hearingList.hearingListMainModel.caseHearings.filter(caseHearing => caseHearing.hearingID === this.hearingId);
+          this.caseHearing = caseHearings.length ? caseHearings[0] : undefined;
+        }
+      });
+    this.hearingChangeReason = this.route.snapshot.data.hearingChangeReason;
+    this.initForm();
+  }
+
+  public get getReasonsTypeFormArray(): FormArray {
+    return this.formBuilder.array(this.hearingChangeReason.map(val => this.formBuilder.group({
+      key: [val.key],
+      value_en: [val.value_en],
+      value_cy: [val.value_cy],
+      hintText_EN: [val.hintText_EN],
+      hintTextCY: [val.hintTextCY],
+      order: [val.order],
+      parentKey: [val.parentKey],
+      selected: [!!val.selected]
+    })));
+  }
+
+  public initForm(): void {
+    this.hearingChangeReasonForm = this.formBuilder.group({
+      reasons: this.getReasonsTypeFormArray,
+    });
+  }
+
+  public isFormValid(action: ACTION): boolean {
+    if (!action || action !== ACTION.VIEW_EDIT_SUBMIT) {
+      return true;
+    } else {
+      this.selectionValid = true;
+      const isReasons = (this.hearingChangeReasonForm.controls.reasons as FormArray).controls
+        .filter(reason => reason.value.selected === true).length > 0;
+      if (!isReasons) {
+        this.validationErrors = [{
+          id: `hearing-option-container`, message: HearingChangeReasonMessages.NOT_SELECTED_A_REASON
+        }];
+        this.selectionValid = false;
+      }
+      return this.selectionValid;
+    }
+  }
+
+  public get hearingChangeReasonMessageEnum() {
+    return HearingChangeReasonMessages;
   }
 
   public executeAction(action: ACTION): void {
-    super.navigateAction(action);
+    if (this.isFormValid(action)) {
+      super.navigateAction(action);
+    }
+  }
+
+  public executeContinue(): void {
+    if (this.isFormValid(ACTION.BACK)) {
+        this.router.navigate(['cases', 'case-details', this.caseId, 'hearings']).then();
+    }
+  }
+
+  public getChosenReasons(): LovRefDataModel[] {
+    const mappedReason: LovRefDataModel[] = [];
+    const reasonChosen = (this.hearingChangeReasonForm.controls.reasons as FormArray).controls
+      .filter(reason => reason.value.selected === true);
+    reasonChosen.forEach(element => {
+      mappedReason.push({
+        key: element.value.key,
+        value_en: element.value.value_en,
+        value_cy: element.value.value_cy,
+        hintText_EN: element.value.hintText_EN,
+        hintTextCY: element.value.hintTextCY,
+        order: element.value.order,
+        parentKey: element.value.parentKey,
+      } as LovRefDataModel);
+    });
```

Comment:
This can be simplified to one forEach
```suggestion
    const reasons = (this.hearingChangeReasonForm.controls.reasons as FormArray).controls
      .forEach(reason => if ( reason.value.selected === true) {
      mappedReason.push({
        key: element.value.key,
        value_en: element.value.value_en,
        value_cy: element.value.value_cy,
        hintText_EN: element.value.hintText_EN,
        hintTextCY: element.value.hintTextCY,
        order: element.value.order,
        parentKey: element.value.parentKey,
      } as LovRefDataModel);
    });
```